packages/lesmis/src/views/MapView.js

```diff
@@ -17,27 +33,137 @@ const Container = styled.div`
   flex: 1;
   height: 100%;
 `;
 
+const EmptyPanel = styled(MapOverlaysPanelContainer)`
+  padding: 2rem;
+`;
+
+const Legend = styled(LegendComponent)`
+  position: absolute;
+  bottom: 5px;
+  z-index: 999;
+  left: 50%;
+  transform: translateX(-50%);
+`;
+
+const TilePicker = styled(TilePickerComponent)`
+  position: absolute;
+  right: 0;
+  bottom: 0;
+  z-index: 9999;
+  pointer-events: none;
+`;
+
+const useProvinceData = () => {
+  const query = useEntitiesData();
+  let provinces = [];
+
+  if (query.data) {
+    provinces = query.data.filter(entity => entity.parentCode === 'LA');
+  }
+
+  return { ...query, data: provinces };
+};
+
 export const MapView = () => {
   const { entityCode } = useUrlParams();
+  const [activeTileSetKey, setActiveTileSetKey] = useState(TILE_SETS[0].key);
+  const [hiddenMeasures, setHiddenMeasures] = useState([]);
   const [selectedOverlay, setSelectedOverlay] = useState(null);
 
+  useEffect(() => {
+    setHiddenMeasures([]);
+  }, [setHiddenMeasures, selectedOverlay]);
+
+  const { data: entityData } = useEntityData({
+    entityCode,
+  });
+
+  const { data: overlayReportData } = useMapOverlayReportData({
+    entityCode,
+    mapOverlayCode: selectedOverlay,
+    hiddenMeasures,
+  });
+
   const { data: overlaysData, isLoading } = useMapOverlaysData({ entityCode });
 
+  const { data: provicesData } = useProvinceData();
+
+  const handleLocationChange = (map, bounds) => {
+    const mapBoxBounds = L.latLngBounds(bounds);
+    const maxBounds = mapBoxBounds.pad(2);
+    map.setMaxBounds(maxBounds);
+  };
+
+  const handleSetHiddenMeasures = value => {
+    const exists = hiddenMeasures.includes(value);
+
+    if (exists) {
+      const newHiddenMeasures = hiddenMeasures.filter(m => m !== value);
+      setHiddenMeasures(newHiddenMeasures);
+    } else {
+      const newHiddenMeasures = [...hiddenMeasures, value];
+      setHiddenMeasures(newHiddenMeasures);
+    }
+  };
+
+  const activeTileSet = TILE_SETS.find(tileSet => tileSet.key === activeTileSetKey);
+  const isCountry = entityData?.type === 'country';
+
   return (
     <Container>
-      <MapOverlaysPanel
-        isLoading={isLoading}
-        overlays={overlaysData}
-        selectedOverlay={selectedOverlay}
-        setSelectedOverlay={setSelectedOverlay}
-      />
-      <Main>Map</Main>
+      {isCountry ? (
+        <EmptyPanel>Select a province from the map to view data.</EmptyPanel>
+      ) : (
+        <MapOverlaysPanel
+          isLoading={isLoading}
+          overlays={overlaysData}
+          selectedOverlay={selectedOverlay}
+          setSelectedOverlay={setSelectedOverlay}
+        />
+      )}
+      <Main>
+        <Map
+          bounds={entityData ? entityData.bounds : null}
+          onLocationChange={handleLocationChange}
+          dragging
+        >
+          <TileLayer tileSetUrl={activeTileSet.url} />
+          {isCountry ? (
+            <PolygonLayer entities={provicesData} Polygon={EntityPolygonLink} />
```

Comment:
If it's the country entity, then display the clickable regions rather than all the overlays because the amount of overlays at laos country level (thousands) jams rendering. This imitates behaviour on tupaia.org